File: 프로그래머스/2/154538. 숫자 변환하기/숫자 변환하기.py
def solution(x, y, n):
    answer = 0
    
    s = set([x])
    while s:
        if y in s: return answer

        ns = set()
        for i in s:
            if i + n <= y:
                ns.add(i + n)
            if i * 2 <= y:
                ns.add(i * 2)
            if i * 3 <= y:
                ns.add(i * 3)
        answer += 1
        s = ns
    
    return -1

File: 프로그래머스/2/154538. 숫자 변환하기/숫자 변환하기.kt
class Solution {
    fun solution(x: Int, y: Int, n: Int): Int {
        var answer: Int = 0
        var s = setOf(x)
        while (s.isNotEmpty()) {
            if (s.contains(y)) return answer
            val ns = mutableSetOf<Int>()
            s.forEach {
                if (it + n <= y) ns.add(it + n)
                if (it * 2 <= y) ns.add(it * 2)
                if (it * 3 <= y) ns.add(it * 3)
            }
            answer++
            s = ns
        }

        return -1
    }
}